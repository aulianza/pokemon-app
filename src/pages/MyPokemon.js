import React, { useState } from "react";
import getOwnedPokemon from "../graphql/queries/getOwnedPokemon";
import releaseOwnedPokemon from "../graphql/operations/removeOwnedPokemon";
import EmptyState from "../components/EmptyState";
import PokemonItem from "../components/PokemonItem";

import styled from "@emotion/styled";

const MyPokemon = () => {
    const [pokemonsState, setPokemonsState] = useState(getOwnedPokemon());

    const releaseHandler = (data) => {
        releaseOwnedPokemon(data);
        setPokemonsState(getOwnedPokemon());
    };

    if (!pokemonsState.length) {
        return (
            <>
                <StyledPageHeader>
                    <h2>Catched Pokemon List</h2>
                    <code>Total: {pokemonsState.length}</code>
                </StyledPageHeader>
                <EmptyState />
            </>
        );
    }

    return (
        <>
            <StyledPageHeader>
                <h2>Catched Pokemon List</h2>
                <code>Total: {pokemonsState.length}</code>
            </StyledPageHeader>
            <StyledPokemonList>
                {pokemonsState.map((poke, index) => (
                    <PokemonItem
                        key={index}
                        poke={poke}
                        releaseAction={true}
                        currentPage="myPokemon"
                        onRelease={releaseHandler}
                    />
                ))}
            </StyledPokemonList>
        </>
    );
};

export default MyPokemon;

const StyledPageHeader = styled.div`
    margin-bottom: 3rem;
`;

const StyledPokemonList = styled.div`
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    grid-row-gap: 1rem;
    row-gap: 1rem;
    column-gap: 1rem;
`;
