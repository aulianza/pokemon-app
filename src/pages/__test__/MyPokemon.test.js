import MyPokemon from '../MyPokemon';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from "react-dom/test-utils";

let container = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("my-pokemon, snapshot", () => {
    act(() => {
        render(<MyPokemon />, container)
    })
    expect(container).toMatchSnapshot();
})