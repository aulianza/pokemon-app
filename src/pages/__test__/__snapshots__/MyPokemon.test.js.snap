// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my-pokemon, snapshot 1`] = `
<div>
  <div
    class="page-header"
  >
    <h2>
      Catched Pokemon List
    </h2>
    <code>
      Total: 
      0
    </code>
  </div>
  <div
    class="empty-state"
  >
    <img
      alt="Not Found"
      height="200"
      src="empty.svg"
      width="200"
    />
    <span>
      No Pokemon Found.
    </span>
  </div>
</div>
`;
